agent/src/control_interface/authorizer.rs

```diff
@@ -0,0 +1,340 @@
+// Copyright (c) 2024 Elektrobit Automotive GmbH
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache License, Version 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0.
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use common::{
+    commands::Request,
+    objects::{AccessRightsRule, ControlInterfaceAccess, ReadWriteEnum},
+    PATH_SEPARATOR,
+};
+
+const WILDCARD_SYMBOL: &str = "*";
+
+#[derive(Clone, Default, Debug, PartialEq)]
+pub struct Authorizer {
+    allow_write_state_rule: Vec<Rule<AllowPathPattern>>,
+    deny_write_state_rule: Vec<Rule<DenyPathPattern>>,
+    allow_read_state_rule: Vec<Rule<AllowPathPattern>>,
+    deny_read_state_rule: Vec<Rule<DenyPathPattern>>,
+    allow_read_write_state_rule: Vec<Rule<AllowPathPattern>>,
+    deny_read_write_state_rule: Vec<Rule<DenyPathPattern>>,
+}
+
+// #[cfg_attr(test, automock)]
+impl Authorizer {
+    #[cfg(test)]
+    pub fn test_value(name: &str) -> Self {
+        Self {
+            allow_write_state_rule: vec![Rule {
+                patterns: vec![AllowPathPattern {
+                    sections: vec![PathPatterSection::String(name.into())],
+                }],
+            }],
+            ..Default::default()
+        }
+    }
+
+    pub fn authorize(&self, request: &Request) -> bool {
```

Comment:
I am currently adapting the control interface and I see that if you have 2-3 rules and more and something is not correctly configured, then the user searches for the reason of the error in the dark. You get back "access denied", which is fine on the workload side itself, but in the debug logs of the agent there isn't a reason mentioned. Maybe we can add debug logs in the agent at least to output something like ("denyRule xY for workload does not match, ....) or something like that, so that a User can find out the reason more easily when looking into the agent logs.

Reply:
On trace level "debug" the agent no logs information why a request has been allowed/denied.

Reply:
I have reviewed and tested it and the logs are helping now if the user does something wrong with the rules.